nixos/modules/services/networking/nftables.nix

```diff
@@ -294,7 +294,7 @@ in
             '') enabledTables)}
             ${cfg.ruleset}
             ${lib.optionalString (cfg.rulesetFile != null) ''
-              include "${cfg.rulesetFile}"
+              ${builtins.readFile cfg.rulesetFile}
```

Comment:
I notice two problems for this implementation:
1. the first problem is the same as https://github.com/NixOS/nixpkgs/pull/230745: `cfg.rulesetFile` can be unavailable at eval time, e.g., `cfg.rulesetFile` is generated by part the of activation script (think `agenix` and `sops-nix`), and if so `builtins.readFile cfg.rulesetFile` fails.
2. even if `cfg.rulesetFile` exists at eval time, e.g., `rulesetFile = "/var/lib/foo"`, one side effect of this change is that the content of `cfg.rulesetFile` is put into the store, which may or may not be desired.

One possible solution for these two problems is to treat `cfg.rulesetFile` differently based on whether `cfg.rulesetFile` is a store path or not.

Reply:
Another problem is that if `cfg.rulesetFile` is not a store path itis not available to the build sandbox.

Reply:
That is to say, if the rulesetFile uses a store path, it can be flattened normally. Otherwise it might not have the permission? But I wondered how can the otherwise method `include` work in this circumstance?

Reply:
> But I wondered how can the otherwise method include work in this circumstance?

Currently, `networking.nftables.preCheckRuleset` is needed to workaround it.

Reply:
I tried using a shell script to extract all files included by rulesetFile and apply the rules for each, but it seems not have the permission also.

Reply:
Sorry, I should have stated that workaround more clearly.  For files generated by the activation script, AFAIK, the only workaround is to delete that `include`.

Reply:
So would it be better to just add a `precheckFlatten` option for this? After all, in the use case like including a rulesetFile and want to change the meta info, it is impossible for the read-only include.

Reply:
I do not understand your use case, could you elaborate it?

Reply:
we can make two options, rulesetFiles and externalRulsetFiles with the first always requiring to be a store path and the second one not being included during checking

Reply:
Ah yeah, seems to be better. Wait me a minute.

Reply:
>  rulesetFiles and externalRulsetFiles with the first always requiring to be a store path and the second one not being included during checking

But will this change breaks some existing usecase? `rulesetFiles` will be flattened default after this change.

Reply:
I think that maybe this is not necessary because we can just include the externalRulesetFiles in the rulesetFile.
Vice, if we just want to use the externalRuleset, leave the rulesetFileFlatten default is ok.